src/uart.rs

```diff
@@ -302,6 +303,116 @@ pub mod config {
         }
     }
 
+    /// Configures the interrupts the UART driver should enable
+    /// in order to be able to quickly inform us about the
+    /// related event.
+    #[derive(Debug, Clone)]
+    pub struct UserInterruptConfig {
```

Comment:
I find the fact that we now have two "interrupt" configurations a bit confusing (even if the first one is interrupt alloc flags). How about naming this `EventConfig`. In the end, it is about configuring what events and when these would be received.